photon-client/src/main/scala/com/linkedin/photon/ml/data/avro/AvroDataReader.scala

```diff
@@ -335,21 +354,36 @@ object AvroDataReader {
     .headOption
 
   /**
+   * Determines whether all the types are numeric
+   *
+   * @param types the types to check
+   * @return true if all specified types are numeric
+   */
+  protected[data] def allNumericTypes(types: Seq[Schema.Type]): Boolean =
+    types.forall(numericPrecedence.toSet)
+
+  /**
+   * Selects the "dominant" avro numeric type from the list. Dominance in this sense means the numeric type with highest
```

Comment:
I think it needs to be more general than size, because `Float` can dominate `Long`, for example.